File: Python/Problems/Problems/problem1423.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1423(Problem):
    def test(self) -> bool:
        cardPoints = [1, 2, 3, 4, 5, 6, 1]
        k = 3

        expected = 12

        result = self.maxScore(cardPoints, k)

        print(result)

        return result == expected

    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)
        curr = 0
        for i in range(k):
            curr += cardPoints[i]
        result = curr
        for i in range(k):
            curr += cardPoints[n - 1 - i] - cardPoints[k - 1 - i]
            result = max(result, curr)

        return result

File: C++/Problems/Problems/problem_1423.cpp
#include "problem_1423.h"

bool problem_1423::test() {
    std::vector cardPoints { 1, 2, 3, 4, 5, 6, 1 };
    int k = 3;

    int expected = 12;

    auto result = maxScore(cardPoints, k);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1423::maxScore(std::vector<int> &cardPoints, int k) {
    int n = cardPoints.size();
    int curr = 0;
    for (int i = 0; i < k; ++i)
        curr += cardPoints[i];
    int result = curr;
    for (int i = 0; i < k; ++i)
    {
        curr += cardPoints[n - 1 - i] - cardPoints[k - 1 - i];
        result = std::max(result, curr);
    }

    return result;
}
